### ISSUE TITLE
Build failure: dino

### ISSUE THREAD

**Issue by toastal** [NixOS/nixpkgs#309206]
### Steps To Reproduce

1. build `dino` on NixOS unstable

### Build log
Goes on & on then fails

```
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incompatible-function-pointer-types
```

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix shell nixpkgs#nix-info --command nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.8, NixOS, 24.05 (Uakari), 24.05.20240502.63c3a29`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.22.0`
 - nixpkgs: `/nix/store/p69bcs7ma6ijj8v9xsrg3nq3nn8ryn95-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

[^1]

[^1]: Please consider [giving up MS GitHub](https://sfconservancy.org/GiveUpGitHub/) or offering a non-proprietary, non-US-corporate-controlled mirror for this free software project. I wish to delete this Microsoft account in the future, but I need more projects like this to support alternative methods to send patches & contribute.

**Comment by eclairevoyant** [comment]
Actual error:
```
/build/source/build/exports/gpgme.vapi:59:23: error: 'struct _gpgme_key' has no member named 'subkeys_length1'
   59 |     public string fpr { get { return subkeys[0].fpr; } }
      |                       ^~
/build/source/plugins/openpgp/src/account_settings_entry.vala: In function 'dino_plugins_open_pgp_account_settings_entry_populate_list_store':
/build/source/plugins/openpgp/src/account_settings_entry.vala:104:30: error: 'struct _gpgme_key' has no member named 'uids_length1'
  104 |             list_store.set(iter, 0, @"$(Markup.escape_text(keys[i].uids[0].uid))\n<span font_family='monospace' font='8'>$(markup_colorize_id(keys[i].fpr, true))</span><span font='8'> </span>");
      |                              ^~
```
```

**Comment by Zonsopkomst** [comment]
Similar, if it is of any help, when I `nixos-rebuild switch --flake .#` I receive the following:
And the full log:
[log.txt](https://github.com/NixOS/nixpkgs/files/15213635/log.txt)
```
error: builder for '/nix/store/jzzl3hp9a811vl246c6xba7a9js978vy-dino-0.4.3.drv' failed with exit code 1;
       last 10 log lines:
       > [850/870] Generating C source src/plugin.c
       > /build/source/plugins/rtp/src/plugin.vala:119.55-119.72: warning: `string.to_uint64' is deprecated. Use uint64.parse
       >   119 |                     stream.on_ssrc_pad_added((uint32) split[4].to_uint64(), pad);
       >       |                                                       ^~~~~~~~~~~~~~~~~~
       > /build/source/plugins/rtp/src/plugin.vala:258.37-258.47: warning: `null' incompatible with return type `Dino.Plugins.Rtp.Stream'
       >   258 |         if (content_params == null) return null;
       >       |                                     ^~~~~~~~~~~
       > Compilation succeeded - 2 warning(s)
       > [851/870] Generating C source src/device.c
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/jzzl3hp9a811vl246c6xba7a9js978vy-dino-0.4.3.drv'.
error: 1 dependencies of derivation '/nix/store/39z6pd695jmjabanlg473snywwx5v2jj-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/07xz1xscq27d1nb2q386nn371qmc6592-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kp7ipcmw2grnms0vsmwpq8qi2ihwdxgg-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/h5mv5fiz84frz11zxps7qcvv2v4dw3jz-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xhmga1xapv9ffypkfsy7fr695vbi6hp4-nixos-system-tyr-24.05.20240503.e9be424.drv' failed to build
```
`nix-shell -p nix-info --run "nix-info -m"`
```
these 6 paths will be fetched (0.75 MiB download, 10.20 MiB unpacked):
  /nix/store/f3ylr5a5w2cla15j2w66n435vwx65xmq-file-5.45
  /nix/store/014nnwb227s2f6zax4nd4v40rjp66582-gnu-config-2023-09-19
  /nix/store/fx2yaklnv3h6jx0p07h1bh4vy618pn9s-gnumake-4.4.1
  /nix/store/ig23ki6sl8dlg3lp0k1l9y9x05wrqs4q-patchelf-0.15.0
  /nix/store/9wnvhjyxjykwn5y06xc9a2h8rs5fbfia-stdenv-linux
  /nix/store/9ckkgn9dakszzcbhxxspjmpwmkp21801-update-autotools-gnu-config-scripts-hook
copying path '/nix/store/014nnwb227s2f6zax4nd4v40rjp66582-gnu-config-2023-09-19' from 'https://cache.nixos.org'...
copying path '/nix/store/f3ylr5a5w2cla15j2w66n435vwx65xmq-file-5.45' from 'https://cache.nixos.org'...
copying path '/nix/store/ig23ki6sl8dlg3lp0k1l9y9x05wrqs4q-patchelf-0.15.0' from 'https://cache.nixos.org'...
copying path '/nix/store/fx2yaklnv3h6jx0p07h1bh4vy618pn9s-gnumake-4.4.1' from 'https://cache.nixos.org'...
copying path '/nix/store/9ckkgn9dakszzcbhxxspjmpwmkp21801-update-autotools-gnu-config-scripts-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/9wnvhjyxjykwn5y06xc9a2h8rs5fbfia-stdenv-linux' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.6-zen1, NixOS, 24.05 (Uakari), 24.05.20240425.7bb2ccd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `""`
 - channels(leeuwarden): `""`
 - nixpkgs: `/nix/store/370qy3d3wg9zhbn5a3dcv6k1q1iigfh4-source`
 ```

**Comment by eclairevoyant** [comment]
Fixed in #309265
